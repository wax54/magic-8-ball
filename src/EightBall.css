.EightBall {
    height: 20rem;
    width: 20rem;
    border-radius: 50%;
    position: relative;
    background-color: black;
}

.EightBall-text{
    font-size: x-large;
    position: absolute;
    text-align: center;
    top: 45%;
    left: 0px;
    right: 0px;
    color: white;
}