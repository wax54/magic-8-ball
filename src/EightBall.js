import { useState } from "react";
import "./EightBall.css";
/** gets a random number between 0 and max (inclusively) */
const randNumber = (max) => Math.floor(Math.random() * (max + 1));

const initialAnswer = {msg: "Think Of a Question", color: "black"};

const defaultAnswers = [
    { msg: "It is certain.", color: "green" },
    { msg: "It is decidedly so.", color: "green" },
    { msg: "Without a doubt.", color: "green" },
    { msg: "Yes - definitely.", color: "green" },
    { msg: "You may rely on it.", color: "green" },
    { msg: "As I see it, yes.", color: "green" },
    { msg: "Most likely.", color: "green" },
    { msg: "Outlook good.", color: "green" },
    { msg: "Yes.", color: "green" },
    { msg: "Signs point to yes.", color: "goldenrod" },
    { msg: "Reply hazy, try again.", color: "goldenrod" },
    { msg: "Ask again later.", color: "goldenrod" },
    { msg: "Better not tell you now.", color: "goldenrod" },
    { msg: "Cannot predict now.", color: "goldenrod" },
    { msg: "Concentrate and ask again.", color: "goldenrod" },
    { msg: "Don't count on it.", color: "red" },
    { msg: "My reply is no.", color: "red" },
    { msg: "My sources say no.", color: "red" },
    { msg: "Outlook not so good.", color: "red" },
    { msg: "Very doubtful.", color: "red" },
];

const EightBall = ({answers=defaultAnswers}) => {
    const [msg, setMsg] = useState(initialAnswer.msg);
    const [color, setColor] = useState(initialAnswer.color);

    const getRandomAnswer = () => answers[randNumber(answers.length - 1)];

    const shake8Ball = () => {
        const newAnswer = getRandomAnswer();
        setMsg(newAnswer.msg);
        setColor(newAnswer.color);
    }

    return (
        <div className="EightBall" 
            style={{ backgroundColor: color }} 
            onClick={shake8Ball}>
            <div className="EightBall-text" >{msg}</div>
        </div>
    )};
export default EightBall