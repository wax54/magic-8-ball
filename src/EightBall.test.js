import React from 'react';
import { render, fireEvent } from '@testing-library/react'; 
import '@testing-library/jest-dom';
import EightBall from './EightBall';

test('it renders without crashing', () => {
    render(<EightBall/>);
});

test('should match snapshot', () => {
    const {asFragment} = render(<EightBall />);
    expect(asFragment()).toMatchSnapshot()
});

test('text should have the right class', () => {
    const { getByText, queryByText } = render(<EightBall />);
    //example of a non exact match
    expect(getByText('Question', { exact: false })).toHaveClass('EightBall-text');

    expect(getByText('Think Of a Question')).toHaveClass('EightBall-text');
    expect(getByText('Think Of a Question')).toBeInTheDocument();
});


test('when eightball clicked, text changes', () => {
    const { getByText, queryByText } = render(<EightBall />);
    const button = getByText('Think Of a Question');
    fireEvent.click(button);
    expect(button).not.toHaveTextContent("Think Of a Question");
});