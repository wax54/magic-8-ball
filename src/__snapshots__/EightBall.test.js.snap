// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="EightBall"
    style="background-color: black;"
  >
    <div
      class="EightBall-text"
    >
      Think Of a Question
    </div>
  </div>
</DocumentFragment>
`;
